d2core/d2inventory/inventory_item_factory.go

```diff
@@ -23,46 +26,88 @@ type InventoryItemFactory struct {
 }
 
 // LoadHeroObjects loads the equipment objects of the hero
-func (f *InventoryItemFactory) loadHeroObjects() {
+func (f *InventoryItemFactory) loadHeroObjects() error {
 	// https://github.com/OpenDiablo2/OpenDiablo2/issues/795
 	//Mode:  d2enum.AnimationModePlayerNeutral.String(),
 	//Base:  "/data/global/chars",
+	shield, err := f.GetArmorItemByCode("buc")
+	if err != nil {
+		return err
+	}
+
+	rhhex, err := f.GetWeaponItemByCode("hax")
+	if err != nil {
+		return err
+	}
+
+	rhwnd, err := f.GetWeaponItemByCode("wnd")
+	if err != nil {
+		return err
+	}
+
+	rhssd, err := f.GetWeaponItemByCode("ssd")
+	if err != nil {
+		return err
+	}
+
+	rhktr, err := f.GetWeaponItemByCode("ktr")
+	if err != nil {
+		return err
+	}
+
+	rhsst, err := f.GetWeaponItemByCode("sst")
+	if err != nil {
+		return err
+	}
+
+	rhjav, err := f.GetWeaponItemByCode("jav")
+	if err != nil {
+		return err
+	}
+
+	rhclb, err := f.GetWeaponItemByCode("clb")
+	if err != nil {
+		return err
+	}
+
 	f.DefaultHeroItems = map[d2enum.Hero]CharacterEquipment{
 		d2enum.HeroBarbarian: {
-			RightHand: f.GetWeaponItemByCode("hax"),
-			Shield:    f.GetArmorItemByCode("buc"),
+			RightHand: rhhex,
+			Shield:    shield,
 		},
 		d2enum.HeroNecromancer: {
-			RightHand: f.GetWeaponItemByCode("wnd"),
+			RightHand: rhwnd,
 		},
 		d2enum.HeroPaladin: {
-			RightHand: f.GetWeaponItemByCode("ssd"),
-			Shield:    f.GetArmorItemByCode("buc"),
+			RightHand: rhssd,
+			Shield:    shield,
 		},
 		d2enum.HeroAssassin: {
-			RightHand: f.GetWeaponItemByCode("ktr"),
-			Shield:    f.GetArmorItemByCode("buc"),
+			RightHand: rhktr,
+			Shield:    shield,
 		},
 		d2enum.HeroSorceress: {
-			RightHand: f.GetWeaponItemByCode("sst"),
-			LeftHand:  f.GetWeaponItemByCode("sst"),
+			RightHand: rhsst,
+			LeftHand:  rhsst,
 		},
 		d2enum.HeroAmazon: {
-			RightHand: f.GetWeaponItemByCode("jav"),
-			Shield:    f.GetArmorItemByCode("buc"),
+			RightHand: rhjav,
+			Shield:    shield,
 		},
 		d2enum.HeroDruid: {
-			RightHand: f.GetWeaponItemByCode("clb"),
-			Shield:    f.GetArmorItemByCode("buc"),
+			RightHand: rhclb,
+			Shield:    shield,
 		},
 	}
+
+	return nil
 }
 
 // GetArmorItemByCode returns the armor item for the given code
-func (f *InventoryItemFactory) GetArmorItemByCode(code string) *InventoryItemArmor {
+func (f *InventoryItemFactory) GetArmorItemByCode(code string) (*InventoryItemArmor, error) {
 	result := f.asset.Records.Item.Armors[code]
 	if result == nil {
-		log.Fatalf("Could not find armor entry for code '%s'", code)
+		return nil, fmt.Errorf("could not find misc item entry for code '%s'", code)
```

Comment:
format string is incorrect, should be:
    could not find __armor__ item entry for code '%s'

Reply:
sorry, copy-paste ;-)